test/acceptance/tests.py

```diff
@@ -923,6 +963,45 @@ def test_staff_override_at_beginning(self):
         ])
 
 
+class FullWorkflowRequiredTest(FullWorkflowBaseTest):
+    """
+    Tests of complete workflows, combining multiple required steps together.
+    """
+    def setUp(self):
+        super(FullWorkflowRequiredTest, self).setUp(problem_loc_key="full_workflow_required")
+
+    @retry()
+    @attr('acceptance')
+    def test_train_self_peer_staff(self):
+        """
+        Scenario: complete workflow that included staff required step.
+
+        Given that I have created a submission, completed training, and done a self assessment
+        And a second learner has also created a submission, training, and self assessment
+        Then I can assess a learner
```

Comment:
These steps aren't correct-- no other learner is assessing this one.

I would recommend 2 different test cases, similar to what was done in the previous "full workflow" tests. In one case, staff grading happens early and peer grading doesn't have to happen to me in order for me to see a score. This is what you have here, though you need to update the BDD description.

In the second case, staff grading happens at the end, after all required steps have been done (including assessment of me by peers). We want to verify that a final grade still doesn't show, until the staff grade is done.

We also need coverage of "Submit and Grade Another Learner" functionality. Although your helper method is written to support that, I don't think any of the tests are guaranteed to actually go through that workflow (you aren't creating multiple submissions in the test case).

As mentioned above, we should have coverage on the ungraded and checked out numbers.

And finally, an accessibility test should be added with the staff grading form in the expanded state.